src/scalaDebugger.ts

```diff
@@ -33,40 +59,37 @@ export function initialize(outputChannel: vscode.OutputChannel): Disposable[] {
 
 export async function start(
   noDebug: boolean,
-  ...parameters: any[]
+  debugParams: DebugDiscoveryParams | ScalaCodeLensesParams
 ): Promise<boolean> {
-  return commands
-    .executeCommand("workbench.action.files.save")
-    .then(() =>
-      vscode.commands.executeCommand<DebugSession>(
-        ServerCommands.DebugAdapterStart,
-        ...parameters
-      )
-    )
-    .then((response) => {
-      if (response === undefined) {
-        return false;
-      }
+  await commands.executeCommand("workbench.action.files.save");
 
-      const port = debugServerFromUri(response.uri).port;
+  const response = await vscode.commands.executeCommand<DebugSession>(
+    ServerCommands.DebugAdapterStart,
+    debugParams
+  );
 
-      const configuration: vscode.DebugConfiguration = {
-        type: configurationType,
-        name: response.name,
-        noDebug: noDebug,
-        request: "launch",
-        debugServer: port, // note: MUST be a number. vscode magic - automatically connects to the server
-      };
-      commands.executeCommand("workbench.panel.repl.view.focus");
-      return vscode.debug.startDebugging(undefined, configuration);
-    });
+  if (response === undefined) {
+    return false;
+  }
+
+  const port = debugServerFromUri(response.uri).port;
+
+  const configuration: vscode.DebugConfiguration = {
+    type: configurationType,
+    name: response.name,
+    noDebug: noDebug,
+    request: "launch",
+    debugServer: port, // note: MUST be a number. vscode magic - automatically connects to the server
+  };
+  commands.executeCommand("workbench.panel.repl.view.focus");
+  return vscode.debug.startDebugging(undefined, configuration);
 }
 
 class ScalaMainConfigProvider implements vscode.DebugConfigurationProvider {
-  resolveDebugConfiguration(
+  async resolveDebugConfiguration(
     _folder: WorkspaceFolder | undefined,
     debugConfiguration: DebugConfiguration
-  ): ProviderResult<DebugConfiguration> {
+  ): Promise<DebugConfiguration | null> {
```

Comment:
Wasn't `ProviderResult` already correct?

Reply:
`async` function has to return `Promise<T>`, `ProviderResult` is a bit weaker than Promise as it's `null | Thenable<T>`